spec/classes/plugin_certguard_spec.rb

```diff
@@ -6,11 +6,12 @@
       let(:facts) { os_facts }
 
       it { is_expected.to compile.with_all_deps }
-      case os
-      when /\-7-/
-         it { is_expected.to contain_package('python3-subscription-manager-rhsm').with(ensure: 'present') }
+      it { is_expected.to contain_pulpcore__plugin('certguard') }
+      if facts[:os]['family'] == 'RedHat' && facts[:os]['release']['major'] == '7'
```

Comment:
The contents of `let` blocks are only available inside `it` blocks so you need to use the `os_facts` variable.
```suggestion
      if os_facts[:os]['family'] == 'RedHat' && os_facts[:os]['release']['major'] == '7'
```